src/ai/backend/common/dto/manager/rbac/response.py

```diff
@@ -0,0 +1,113 @@
+
+from ai.backend.common.api_handlers import BaseResponseModel
+
+from .types import RoleSource, RoleStatus
+
+__all__ = (
+    "RoleDTO",
+    "AssignedUserDTO",
+    "PaginationInfo",
+    "CreateRoleResponse",
+    "GetRoleResponse",
+    "UpdateRoleResponse",
+    "DeleteRoleResponse",
+    "SearchRolesResponse",
+    "AssignRoleResponse",
+    "RevokeRoleResponse",
+    "SearchUsersAssignedToRoleResponse",
+)
+
+
+class RoleDTO(BaseModel):
+    """DTO for role data."""
+
+    id: UUID = Field(description="Role ID")
+    name: str = Field(description="Role name")
+    source: RoleSource = Field(description="Role source")
+    status: RoleStatus = Field(description="Role status")
+    created_at: datetime = Field(description="Creation timestamp")
+    updated_at: Optional[datetime] = Field(default=None, description="Last update timestamp")
+    deleted_at: Optional[datetime] = Field(default=None, description="Deletion timestamp")
+    description: Optional[str] = Field(default=None, description="Role description")
+
+
+class AssignedUserDTO(BaseModel):
+    """DTO for user assigned to a role."""
+
+    user_id: UUID = Field(description="User ID")
+    granted_by: Optional[UUID] = Field(default=None, description="ID of user who granted this role")
+    granted_at: datetime = Field(description="Timestamp when the role was granted")
+
+
+class PaginationInfo(BaseModel):
+    """Pagination information."""
+
+    total: int = Field(description="Total number of items")
+    offset: int = Field(description="Number of items skipped")
+    limit: Optional[int] = Field(default=None, description="Maximum items returned")
+
+
+class CreateRoleResponse(BaseResponseModel):
+    """Response for creating a role."""
+
+    role: RoleDTO = Field(description="Created role")
+
+
+class GetRoleResponse(BaseResponseModel):
+    """Response for getting a role."""
+
+    role: RoleDTO = Field(description="Role data")
+
+
+class UpdateRoleResponse(BaseResponseModel):
+    """Response for updating a role."""
+
+    role: RoleDTO = Field(description="Updated role")
+
+
+class DeleteRoleResponse(BaseResponseModel):
+    """Response for deleting a role."""
+
+    deleted: bool = Field(description="Whether the role was deleted")
+
+
+class SearchRolesResponse(BaseResponseModel):
+    """Response for searching roles."""
+
+    roles: list[RoleDTO] = Field(description="List of roles")
+    pagination: PaginationInfo = Field(description="Pagination information")
+
+
+class AssignRoleResponse(BaseResponseModel):
+    """Response for assigning a role to a user."""
+
+    user_id: UUID = Field(description="User ID")
+    role_id: UUID = Field(description="Role ID")
+    granted_by: Optional[UUID] = Field(default=None, description="ID of user who granted the role")
+
+
+class RevokeRoleResponse(BaseResponseModel):
+    """Response for revoking a role from a user."""
+
+    user_role_id: UUID = Field(description="ID of the revoked user-role association")
```

Comment:
It doesn't seem like the `user_` prefix is needed, please check if it should be included.